FrontEnd/src/HomePage/MainPage.jsx

```diff
@@ -1,7 +1,42 @@
+import {useNavigate} from "react-router-dom";
+import { useEffect, useState } from 'react';
+import { getUID, getUserData } from '../UserAuthentication/FirestoreDB';
+import { logOutUser } from '../UserAuthentication/Auth';
+
 function MainPage(){
+  const navigate = useNavigate();
+  const userUID = getUID();
+  const [userData, setUserData] = useState(null)
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (userUID) {
+        const data = await getUserData(userUID);
+        setUserData(data);
+      }
+    };
+    fetchData();
```

Comment:
You can define this function outside of `useEffect` hook to make the logic more clear. 

```
const fetchData = async () => {
      if (userUID) {
        const data = await getUserData(userUID);
        setUserData(data);
      }
    };

useEffect(() => {
    fetchData();
```